src/lib/menu/menu-trigger.ts

```diff
@@ -260,48 +282,110 @@ export class MdMenuTrigger implements AfterViewInit, OnDestroy {
+      // to the edges of the trigger, instead of overlapping it.
+      overlayFallbackX = originX = this.menu.xPosition === 'before' ? 'start' : 'end';
+      originFallbackX = overlayX = originX === 'end' ? 'start' : 'end';
+
+      // TODO(crisbeto): this should be a function, once the overlay supports it.
+      // Right now it will be wrong for the fallback positions.
+      offsetY = overlayY === 'bottom' ? MENU_PANEL_TOP_PADDING : -MENU_PANEL_TOP_PADDING;
+    } else if (!this.menu.overlapTrigger) {
       originY = overlayY === 'top' ? 'bottom' : 'top';
-      fallbackOriginY = fallbackOverlayY === 'top' ? 'bottom' : 'top';
+      originFallbackY = overlayFallbackY === 'top' ? 'bottom' : 'top';
     }
 
     return this._overlay.position()
-      .connectedTo(this._element,
-          {originX: posX, originY: originY}, {overlayX: posX, overlayY: overlayY})
+      .connectedTo(this._element, {originX, originY}, {overlayX, overlayY})
+      .withDirection(this.dir)
+      .withOffsetY(offsetY)
       .withFallbackPosition(
-          {originX: fallbackX, originY: originY},
-          {overlayX: fallbackX, overlayY: overlayY})
+          {originX: originFallbackX, originY},
+          {overlayX: overlayFallbackX, overlayY})
       .withFallbackPosition(
-          {originX: posX, originY: fallbackOriginY},
-          {overlayX: posX, overlayY: fallbackOverlayY})
+          {originX, originY: originFallbackY},
+          {overlayX, overlayY: overlayFallbackY})
       .withFallbackPosition(
-          {originX: fallbackX, originY: fallbackOriginY},
-          {overlayX: fallbackX, overlayY: fallbackOverlayY});
+          {originX: originFallbackX, originY: originFallbackY},
+          {overlayX: overlayFallbackX, overlayY: overlayFallbackY});
   }
 
+  /** Cleans up the active subscriptions. */
   private _cleanUpSubscriptions(): void {
     if (this._backdropSubscription) {
       this._backdropSubscription.unsubscribe();
     }
+
     if (this._positionSubscription) {
       this._positionSubscription.unsubscribe();
     }
+
+    this._parentMenuSubscriptions.forEach(subscription => subscription.unsubscribe());
+    this._parentMenuSubscriptions = [];
   }
 
+  /** Sets up the subscriptions to parent menu events. */
+  private _subscribeToParentMenu(): void {
+    if (!this.triggersSubmenu()) {
+      return;
+    }
+
+    // Subscribe to changes in the hovered item in order to toggle the panel.
+    this._parentMenuSubscriptions.push(this._parentMenu.hover().subscribe(active => {
+      if (active === this._menuItemInstance) {
+        this._openedByMouse = true;
+        this.openMenu();
+      } else {
+        this.menu.close.emit();
+      }
+    }));
+
+    // Subscribe to the parent menu closing, allowing for the close event
+    // to propagate to any child menus that the current trigger may have.
+    this._parentMenuSubscriptions.push(this._parentMenu.close.subscribe(() => {
+      this.menu.close.emit();
+    }));
+  }
```

Comment:
I did it initially, because they subscribe at different times in the lifecycle. Looking at it again, they don't really need to do that. Refactored.